utils/watchcat/files/watchcat.sh

```diff
@@ -45,29 +46,98 @@ reboot_now() {
 }
 
 watchcat_periodic() {
-	local period="$1"
-	local force_delay="$2"
+	local failure_period="$1"
+	local force_reboot_delay="$2"
 
-	sleep "$period" && reboot_now "$force_delay"
+	sleep "$failure_period" && reboot_now "$force_reboot_delay"
+}
+
+watchcat_restart_modemmanager_iface() {
+	[ $2 -gt 0 ] && {
+		logger -t INFO "Resetting current-bands to 'any' on modem: $1 now."
+		/usr/bin/mmcli -m any --set-current-bands=any
+	}
+	logger -t INFO "Reconnecting modem: $1 now."
+	/etc/init.d/modemmanager restart
+	ifup "$1"
+}
+
+watchcat_restart_network_iface() {
+	logger -t INFO "Restarting network interface: $1."
+	ifdown "$1"
+	ifup "$1"
+}
+
+watchcat_monitor_network() {
+	local failure_period="$1"
+	local ping_hosts="$2"
+	local ping_frequency_interval="$3"
+	local ping_size="$4"
+	local iface="$5"
+	local mm_iface_name="$6"
+	local mm_iface_unlock_bands="$7"
+
+	local time_now="$(cat /proc/uptime)"
+	time_now="${time_now%%.*}"
+
+	[ "$time_now" -lt "$failure_period" ] && sleep "$((failure_period - time_now))"
+
+	time_now="$(cat /proc/uptime)"
+	time_now="${time_now%%.*}"
+	local time_lastcheck="$time_now"
+	local time_lastcheck_withinternet="$time_now"
+
+	local ping_size="$(get_ping_size "$ping_size")"
+
+	while true; do
+		# account for the time ping took to return. With a ping time of 5s, ping might take more than that, so it is important to avoid even more delay.
+		time_now="$(cat /proc/uptime)"
+		time_now="${time_now%%.*}"
+		local time_diff="$((time_now - time_lastcheck))"
+
+		[ "$time_diff" -lt "$ping_frequency_interval" ] && sleep "$((ping_frequency_interval - time_diff))"
+
+		time_now="$(cat /proc/uptime)"
+		time_now="${time_now%%.*}"
+		time_lastcheck="$time_now"
+
+		for host in $ping_hosts; do
+			if ping -I "$iface" -s "$ping_size" -c 1 "$host" &>/dev/null; then
+				time_lastcheck_withinternet="$time_now"
+			else
+				logger -p daemon.info -t "watchcat[$$]" "no internet connectivity on "$iface" for $((time_now - time_lastcheck_withinternet)). Restarting "$iface" after reaching $failure_period"
+			fi
+		done
+
+		[ "$((time_now - time_lastcheck_withinternet))" -ge "$failure_period" ] && {
+
+			if [ "$mm_iface_name" -ne "" ]; then
```

Comment:
inverted logic?

Reply:
Thanks for catching that, fixed. Wrong comparison operator for strings.